elixir/apps/web/lib/web/live/actors/show.ex

```diff
@@ -359,4 +390,9 @@ defmodule Web.Actors.Show do
       identity -> identity.last_seen_at
     end
   end
+
+  defp has_email?(identity) do
```

Comment:
I think we can make code a bit cleaner by moving this function below `get_identity_email(identity)` in core components to keep things together, and then we can simplify the code a bit too:
```
  def get_identity_email(identity) do
    provider_email(identity) or identity.provider_identifier
  end

  def identity_has_email?(identity) do
    not is_nil(provider_email(identity)) or identity.provider.adapter == :email
  end

  defp provider_email(identity) do
    get_in(identity.provider_state, ["userinfo", "email"])
  end
```

Reply:
Yeah, that sounds good 👍